settings: Improve settings info for servers that have not enabled mobile push notifications.

sayamsamal opened the pull request:
<!-- Describe your pull request here.-->
**Tasks Tracker:**

Changes to apply to:
`Stream settings`
`Personal settings > Notifications (top section)`
`Default user settings > Notification triggers`
- [x]  Change the tooltip to: "Mobile push notifications are not enabled on this server." 
- [x]  Move the tooltip to the disabled setting itself, rather than a question mark after it. This question mark icon should be removed. In the triggers tables, ideally, both the "Mobile" heading and the checkboxes themselves would have this tooltip.
- [x]  Add a question mark help center link (with regular styling), pointing to https://zulip.com/help/mobile-notifications#enabling-push-notifications-for-self-hosted-servers.

Changes apply to:
`Personal settings > Notifications > Mobile message notifications`
`Default user settings > Mobile message notifications`
- [x] Add a yellow banner at the top of the section: "Mobile push notifications are not enabled on this server. [Learn more](https://zulip.com/help/mobile-notifications#enabling-push-notifications-for-self-hosted-servers)"

Fixes: #27830

CZO Discussion: [Link](https://chat.zulip.org/#narrow/stream/101-design/topic/mobile.20push.20notifications.20improve.20settings.20info.20.2327830/near/1687950)

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
- Change the tooltip to: "Mobile push notifications are not enabled on this server." 
![image](https://github.com/zulip/zulip/assets/82862779/2f57a449-fe1e-44f2-b45d-05c80a20fd2d)
- Move the tooltip to the disabled setting itself, rather than a question mark after it.
![image](https://github.com/zulip/zulip/assets/82862779/1b4f5ce5-43d9-4834-83e0-2665b5c1318a)
- In the triggers tables, ideally, both the "Mobile" heading and the checkboxes themselves would have this tooltip.
![image](https://github.com/zulip/zulip/assets/82862779/4f55492e-88b7-48e5-9f99-d50ec7d79377)
![image](https://github.com/zulip/zulip/assets/82862779/763b7eed-44b9-4b9b-8ab5-9eea517f7e90)
- Add a question mark help center link (with regular styling), pointing to https://zulip.com/help/mobile-notifications#enabling-push-notifications-for-self-hosted-servers.
![image](https://github.com/zulip/zulip/assets/82862779/f0ef3530-4b1a-4392-8b26-0bb317a42d2b)
- Add a yellow banner at the top of the section: "Mobile push notifications are not enabled on this server. [Learn more](https://zulip.com/help/mobile-notifications#enabling-push-notifications-for-self-hosted-servers)"
![image](https://github.com/zulip/zulip/assets/82862779/9961296c-1a31-4d09-a2cc-3ca2879336e7)
![image](https://github.com/zulip/zulip/assets/82862779/3e47b632-e05a-470d-8ec9-289e537e0255)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
We should put this banner below the section heading, rather than above.

<img width="720" alt="Screenshot 2023-11-27 at 5 01 11 PM" src="https://github.com/zulip/zulip/assets/2090066/6735f357-7adf-40e7-bdab-12da492dcee8">

alya commented:
Let's also disable changing the " Send mobile notifications even if I'm online" setting in this situation. It's weird to have that one enabled while the rest are disabled, I think.

alya commented:
Let's make the question mark (1) not look disabled, since it's a working link users can click on, and (2) not trigger the tooltip, since I think that makes it harder to recognize it as a help center link.


<img width="101" alt="Screenshot 2023-11-27 at 5 03 03 PM" src="https://github.com/zulip/zulip/assets/2090066/460ca365-411f-4db9-83fe-b34df4ed6f24">

sayamsamal commented:
@alya I have addressed your feedback and pushed the new code. Could you please review this PR again? Attaching the new screenshots below: 
- Help link widget not disabled
![image](https://github.com/zulip/zulip/assets/82862779/5afa2475-fed2-443f-8984-846d55260249)
- No "Mobile push notifications are not enabled on this server" tooltip over the help link widget
![image](https://github.com/zulip/zulip/assets/82862779/c6be94a0-6ceb-42bf-a6e4-7f569778afc9)
- Tooltip over the label + checkbox
![image](https://github.com/zulip/zulip/assets/82862779/f713a563-3ab2-465a-9df6-c67adaf1d4de)
- Similar behavior on Notification Triggers tooltips.
![image](https://github.com/zulip/zulip/assets/82862779/150bc50c-ea5e-4847-81d9-b64447f04be3)
- Banner inside Mobile message notifications sub-section + related option disabled.
![image](https://github.com/zulip/zulip/assets/82862779/1c8f4a07-c831-47e8-ab90-9e595e0c9373)

alya commented:
This looks close! The banner should be as wide as the text, not full-width. The vertical spacing around it also doesn't look quite right. For reference:

<img width="627" alt="Screenshot 2023-11-30 at 9 51 46 PM" src="https://github.com/zulip/zulip/assets/2090066/ceebbdf6-681e-4999-a779-3e6ac3e910a9">

alya commented:
Everything else looked good to me!

sayamsamal commented:
Pushed updated code in which the banner fits the width of the text inside.
![image](https://github.com/zulip/zulip/assets/82862779/3af05de5-32ea-4ad2-a2fa-9fc3c22616a2)

sayamsamal commented:
Fixed vertical spacing issue.
![image](https://github.com/zulip/zulip/assets/82862779/e6aaf4f9-26cb-4c14-ace3-66aaab8cece9)

sahil839 reviewed (commented):
Posted a few minor comments but looks good to me overall.

sayamsamal commented on web/templates/settings/notification_settings.hbs:
```diff
@@ -128,7 +129,6 @@
     </div>
 
     <div class="mobile_notifications m-10 {{#if for_realm_settings}}settings-subsection-parent{{else}}subsection-parent{{/if}}">
-
```
Reverted unnecessary changes.

sayamsamal commented on web/styles/settings.css:
```diff
@@ -43,6 +43,7 @@ h3,
 
     &:hover {
         opacity: 1;
+        color: inherit !important;
```
Addressed by increasing the specificity by changing the selector from `.help_link_widget` (0, 1, 0) to `a.help_link_widget` (0, 1, 1).

sayamsamal commented on web/templates/settings/notification_settings.hbs:
```diff
@@ -148,6 +148,7 @@
               is_checked=(lookup ../settings_object this)
+              is_disabled=(lookup ../show_push_notifications_tooltip this)
               label=(lookup ../settings_label this)
               show_push_notifications_tooltip=(lookup ../show_push_notifications_tooltip this)
```
Added commit 334fbb08341af10d5e1b039aa5f64d257e1647d1 to remove the `show_push_notifications_tooltip` param.

sayamsamal commented:
Thank you for the review @sahil839! I have tried my best to address your feedback and have mentioned the changes in the comments above.

timabbott commented:
Merged, thanks @sayamsamal! 

Thought I just noticed a bug: In the dark theme, the `help_link_widget` still turns blue on hover. Can you do a quick follow-up to address that?

sayamsamal commented:
Opened #28064, as a follow-up to address the blue help link on hover in dark mode. Thanks for the merge! 🎉
